sklearn/metrics/classification.py

```diff
@@ -186,6 +186,69 @@ def accuracy_score(y_true, y_pred, normalize=True, sample_weight=None):
     return _weighted_sum(score, sample_weight, normalize)
 
 
+def balanced_accuracy_score(y_true, y_pred):
+    """Balanced accuracy score
+
+    The balanced accuracy score is defined as
+    0.5*true positives/(true positives + false negatives) +
+    0.5*true negatives/(true negatives + false positives)
+
+    This function is equal to the average of positive label recall
+    and negative label recall.
+
+    Parameters
+    ----------
+    y_true : array, shape = [n_samples]
+        Ground truth (correct) target values.
+
+    y_pred : array, shape = [n_samples]
+        Estimated targets as returned by a classifier.
+
+    Returns
+    -------
+    C : array, shape = [n_classes, n_classes]
+        Confusion matrix
+
+
+    References
+    ----
+    .. [1] `Wikipedia entry for balanced accuracy
+    http://en.wikipedia.org/wiki/Accuracy_and_precision#In_binary_classification
+
+    Examples
+    --------
+    >>> from sklearn.metrics import balanced_accuracy_score
+    >>> y_true = [0, 0, 1, 1]
+    >>> y_pred = [0, 1, 1, 1]
+    >>> balanced_accuracy_score(y_true, y_pred)
+    0.75
+
+    >>> y_true = [0,1,1,1,1]
+    >>> y_pred = [1,1,1,1,1]
+    >>> balanced_accuracy_score(y_true, y_pred)
+    0.5
+
+    >>> y_true = ['b','a','a','a']
+    >>> y_pred = ['a','a','b','a']
+    >>> balanced_accuracy_score(y_true, y_pred)
+    0.33333333333333331
```

Comment:
Please use `# doctest: +ELLIPSIS` and `0.333...` to make this doctests robust to platform specific floating point representations.